openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -641,107 +641,108 @@ def password_change_request_handler(request):
         return HttpResponseBadRequest(_("No email address provided."))
 
 
-@require_POST
-@ensure_csrf_cookie
-def password_reset_token_validate(request):
-    """HTTP end-point to validate password reset token. """
-    is_valid = False
-    token = request.POST.get('token')
-    try:
-        token = token.split('-', 1)
-        uid_int = base36_to_int(token[0])
-        if request.user.is_authenticated and request.user.id != uid_int:
-            return JsonResponse({'is_valid': is_valid})
+class PasswordResetTokenValidation(APIView):
 
-        user = User.objects.get(id=uid_int)
-        if UserRetirementRequest.has_user_requested_retirement(user):
-            return JsonResponse({'is_valid': is_valid})
+    @method_decorator(ensure_csrf_cookie)
+    def post(self, request):
+        """ HTTP end-point to validate password reset token. """
+        is_valid = False
+        token = request.data.get('token')
+        try:
+            token = token.split('-', 1)
+            uid_int = base36_to_int(token[0])
+            if request.user.is_authenticated and request.user.id != uid_int:
+                return JsonResponse({'is_valid': is_valid})
 
-        is_valid = default_token_generator.check_token(user, token[1])
-        if is_valid and not user.is_active:
-            user.is_active = True
-            user.save()
-    except Exception:   # pylint: disable=broad-except
-        AUDIT_LOG.exception("Invalid password reset confirm token")
+            user = User.objects.get(id=uid_int)
+            if UserRetirementRequest.has_user_requested_retirement(user):
+                return JsonResponse({'is_valid': is_valid})
 
-    return JsonResponse({'is_valid': is_valid})
+            is_valid = default_token_generator.check_token(user, token[1])
+            if is_valid and not user.is_active:
+                user.is_active = True
+                user.save()
+        except Exception:   # pylint: disable=broad-except
+            AUDIT_LOG.exception("Invalid password reset confirm token")
 
+        return JsonResponse({'is_valid': is_valid})
 
-def _check_token_has_required_values(uidb36, token):
-    """
-    Helper function to test that token
-    string passed has the required kwargs needed
-    to process token validation.
-    """
 
-    if not uidb36 or not token:
-        return False, None
-    try:
-        uid_int = base36_to_int(uidb36)
-    except ValueError:
-        return False, None
-    return True, uid_int
+class LogistrationPasswordReset(APIView):
```

Comment:
nit: change view name `PasswordResetView`, also add a docstring explaining what this view does.